bijection-hbase/src/test/scala/com/twitter/bijection/hbase/HBaseInjectionsLaws.scala

```diff
@@ -0,0 +1,104 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.twitter.bijection.hbase
+
+import org.scalatest.{ PropSpec, MustMatchers }
+import org.scalatest.prop.PropertyChecks
+import org.scalacheck.Arbitrary
+
+import com.twitter.bijection.{ CheckProperties, BaseProperties, Injection }
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable
+
+object HBaseInjectionsLaws {
+  implicit val arbitaryImmutableBytesWritable: Arbitrary[ImmutableBytesWritable] =
+    Arbitrary(for {
+      arbByteArray <- Arbitrary.arbitrary[Array[Byte]]
+    } yield new ImmutableBytesWritable(arbByteArray))
+
+  implicit val equivImmutableBytesWritable = new Equiv[ImmutableBytesWritable] {
+    def equiv(x: ImmutableBytesWritable, y: ImmutableBytesWritable): Boolean = x == y
+  }
+}
+
+/**
+ * @author Muhammad Ashraf
+ * @since 7/10/13
+ */
+class HBaseInjectionsLaws extends CheckProperties with BaseProperties {
+  import HBaseInjections._
+
+  import HBaseInjectionsLaws.arbitaryImmutableBytesWritable
+  import HBaseInjectionsLaws.equivImmutableBytesWritable
+
+  property("String -> Array[Byte]") {
```

Comment:
These `Array[Byte]` injections may be superfluous. Their purpose is for the situation where you have a value serialized as `Array[Byte]` with hbase encoding. I would need to know more about how people use hbase before I'm confident in removing them.

Perhaps you're speaking to the lack of the `Rep` type? I guess it's possible that these injections may enable users to make mistakes by growing the domain from `Array[Byte] @@ Rep[T]` to `Array[Byte]`.